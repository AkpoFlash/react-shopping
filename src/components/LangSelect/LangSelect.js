import React from 'react';
import { connect } from 'react-redux';
import map from 'lodash/map';

import { t } from '../../helpers';
import LANG from  '../../configs/languages';
import { setLanguage } from '../../actions/languages';

const LangSelect = React.memo((props) => {
	
	const handleChangeLang = (e) => {
		props.setLanguage({
			type: e.currentTarget.value,
		});
	}

	return (
		<form >
			<select value={props.usersLang} onChange={ handleChangeLang }>
			{
				map(LANG, (item, i) => {
					return (
						<option 
							value={ item.type } 
							key={ item.type } >
							{t(item.type)}
						</option>
					);
				})
			}
			</select>
		</form>
	);
});

const mapStateToProps = ({ languages }) => ({
	usersLang: languages.usersLang,
});

const mapDispatchToProps = (dispatch) => ({
	setLanguage: ( lang ) => dispatch(setLanguage(lang)),
});

export default connect(mapStateToProps, mapDispatchToProps)(LangSelect);
