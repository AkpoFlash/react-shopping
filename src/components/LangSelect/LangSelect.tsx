import * as React from 'react';
import { connect } from 'react-redux';
import { map } from 'lodash';
import * as PropTypes from 'prop-types';

import { t } from '../../helpers';
import LANG from  '../../configs/languages';
import { setLanguage } from '../../actions/languages';

export const LangSelect: React.FunctionComponent<any> = (props) => {
	
	const handleChangeLang = (e) => {
		props.setLanguage({
			type: e.currentTarget.value,
		});
	}

	return (
		<form >
			<select value={props.usersLang} onChange={ handleChangeLang }>
			{
				map(LANG, (item, i) => {
					return (
						<option 
							value={ item.type } 
							key={ item.type } >
							{t(item.type)}
						</option>
					);
				})
			}
			</select>
		</form>
	);
};

const mapStateToProps = ({ languages }) => ({
	usersLang: languages.usersLang,
});

const mapDispatchToProps = (dispatch) => ({
	setLanguage: ( lang ) => dispatch(setLanguage(lang)),
});

LangSelect.propTypes = {
	usersLang: PropTypes.string.isRequired,
	setLanguage: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(React.memo(LangSelect));
