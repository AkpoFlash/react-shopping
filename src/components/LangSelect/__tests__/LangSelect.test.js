import React from 'react';
import { render } from 'enzyme';

import { LangSelect } from '../LangSelect';

describe('LangSelect', () => {
	const langSelect = render(<LangSelect />);

	it('check LangSelect snapshot', () => {
		expect(langSelect).toMatchSnapshot();
	});
});