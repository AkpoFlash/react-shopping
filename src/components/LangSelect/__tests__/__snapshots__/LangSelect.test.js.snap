// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LangSelect check LangSelect snapshot 1`] = `
<form>
  <select>
    <option
      value="Eng"
    >
      Eng
    </option>
    <option
      value="Rus"
    >
      Rus
    </option>
  </select>
</form>
`;
